### ISSUE TITLE
DVB Player subitles speedup or slowdown

### ISSUE THREAD

**Issue by YoKo666** [mx3L/subssupport#10]
Can you please add this support to DVBPlayer Subsupport? So we can fine-tune subtitles.. LEFT-RIGHT arrows move "one title" ahead or back, but how about adding option to slow/speed up for 0.5 sec (key 4 or 6) and +-1.0 sec with keys 1 and 3
If the option is there it doesn't work, at least for DVBPlayer.
Also, a small cosmetic change.. Option so we can change distance between 2 lines of titles (when there are 2 or more lines on screen at the time). By doing this we can be able to split background so it doesn't have to be at one piece.
Adding an option of background to be 0px out of fonts will be also useful.
Thanx in advance..Best subtitler there is :+1:

**Comment by mx3L** [comment]
Thanks for very good suggestions.

Currently working on fine-tuning:
I've implemented this by doing seeking to precise time. Currently I'm thinking about binding it to up/down button:

UP: seek +200ms
DOWN: seek -200ms
UP/DOWN for restartSubtitle moved to OK
OK for showing status, moved to INFO

Have to think about how to show this operation in GUI, will send here some test version once done.

**Comment by YoKo666** [comment]
Hello there and thank you for your reply.
If you ask me i wouldn't touch OK button in DVB player. Mind that some boxes (ie. VU+) don't have INFO button on remote, so it would be impossible to see status.Maybe if you add it to [TV] button?
As for lines, when there are more than 2 lines of text DVB player doesn't show it correctly on screen. It shows 3 lines in 2, what makes a mess to read because you see bottom of first line, you see the whole 2nd line and you see only top of the 3rd line of text in 2 rows. Impossible to read and really not an eye candy :) Looks like DVB player isn't capable of reproducing more than 2 lines of text on screen. Otherwise, as i can say (not a noob) best subtitler can be even better with this :)

**Comment by YoKo666** [comment]
Any progress please?

**Comment by YoKo666** [comment]
Can we expect new version please?
Thank you!

**Comment by mx3L** [comment]
Hi didn't get to finish it yet, hopefully this week..

**Comment by YoKo666** [comment]
That's wonderful news! Looking forward to it. Best subtitler so far, with this new option will be even better.

**Comment by mx3L** [comment]
Hi, 

1. EPG button on VU should be the same as INFO button elsewhere.

2. I don't see problem with 3 lines:
![3lines](https://cloud.githubusercontent.com/assets/3874649/14249841/ff175a9a-fa7b-11e5-898a-e6a2b0a9c737.jpg)

Can you test this sample srt file, how it shows 3 lines for you?
[test_tags.zip](https://github.com/mx3L/subssupport/files/202528/test_tags.zip)

**Comment by YoKo666** [comment]
Yes, seems to be OK, guess something was wrong with the subtitle encoding then with my downloaded sub.
Can you notice the bug with italic/bold/regular at same line on your picture? It shows all italic, even the first line is BOLD, second ITALIC and third REGULAR? Seems to me that Italic has highest priority in subitltes and then it changes the whole lines to italic. Can this be fixed also?
Looking forward to new version :)

**Comment by YoKo666** [comment]
Any news please?

**Comment by mx3L** [comment]
Hello, sorry I didn't finish GUI part, thought these mini seeks are now possible, please try it if it works as it should. UP/DOWN are now for fine adjustments.
https://dl.dropboxusercontent.com/u/38760017/enigma2-plugin-extensions-subssupport_1.5.4-20160215_all.ipk

**Comment by YoKo666** [comment]
It doesn't work, now and UP and DOWN have same function, reloads sub at line start time :(

**Comment by mx3L** [comment]
I double checked contents of subssupport which I posted you and it's all right, so UP/DOWN is changing subtitles time by +-200ms.
So It looks like you have still old version there.

Look at the picture above which I posted. In new version "Subtitles Time" also shows miliseconds. If you see miliseconds then it's installed properly, now you're seeing probably only seconds.

**Comment by YoKo666** [comment]
Hmm..you're right, my box says _ not downgrading previous version (which is the one i've downloaded from here before). So, the timestamp is not right then on your new version looks like. Anyway i will try to delete completely older version and install new one then i will report.

**Comment by mx3L** [comment]
I had to change versioning, because of creating now also debian package. That's why is telling you that what are you trying to install is older version.

Just use `opkg --force-downgrade install enigma2-plugin-extensions-subssupport...`

**Comment by YoKo666** [comment]
Yes, i can confirm it is working flawless!! Thanx, it's the real thing that was missing now it's ok! SubSupport getting better and better :+1: 
Anyway the only flaw is you need to uninstall older version first coz of wrong timestamp or something. On my box it wouldn't want to upgrade over older version enigma2-plugin-extensions-subssupport_1.5.4_20151105_all.ipk so,you might want to look into it also :)
Thank you once more for this great sub-app :)

**Comment by mx3L** [comment]
Glad to hear that :), I'm still not sure how should it appear in user interface when someone presses UP/DOWN, is it good to just show status bar, where you can see that by pressing these buttons is actually subtitles time changing?

I was thinking, that maybe instead of showing status bar, there should shown label somewhere with operation name, i.e when presses UP, on screen there will be shown "+200ms" and then it will fade/change color if you don't press again, I think it would be cleaner. 
Or maybe you have better suggestion?

**Comment by YoKo666** [comment]
I think maybe an option in settings with 3 options:
- don't show info bar (suits best coz the bar is too big)
- show info bar +200ms/-200ms (maybe with sub option of timing 1-5sec if not, then fading is ok, but make it timed to 1-3sec else it will cover too much of the screen or it will be annoying if  stays too long)
- show full info bar (says it all)
What do you say?
Also, how's the progress with fixing italic font has the priority over regular and bold? (As shown on your screenshot).

**Comment by YoKo666** [comment]
Found few bugs in your new version 1.5.4-20160215 . First, the buttons < and > (not on a wheel with UP and DOWN, but those on numpad) doesn't work anymore. Those were for skipping few minutes of titles (forward/backward). Now the buttons reset subs to 0:00. 
No option to enter direct time of subtitle jump..
Also, if you put option to sync event, if the time is higher than 30 or more minutes for position the titles, it crashes E2.
Can you please look into it? Otherwise HELP button crashes E2 on Vu+ is another bug.

**Comment by YoKo666** [comment]
Found another bug, and this one's big!
With the new version 1.5.4-20160215, when i speed the sub with UP or DOWN, after some time (usually 5-7 minutes) the subs goes out of sync (it just doesn't match anymore, subs goes to +-1sec or even more), so after every 5-7 minutes the subs has to be speeded or slowed again. I've tested SAME sub with old version (uninstalled newer) and the subs are ok (as i sync them at the beginning they stayed in sync for the whole movie), no need to readjust.So i think this version broke the timing of the subs!! 
Another bug is when i press UP or DOWN when no lines on screen it jumps to the next line and not +-200ms, it then f**ks up the whole timing.
I've reverted to the old version, and it's more stable with all the bugs polished (except Italic takeover priority over regular or bold font)..

**Comment by mx3L** [comment]
Thanks for letting me know, it's rewritten almost from scratch with not much testing done, so it's certainly possible that there are bugs in there. I will try to fix it.

**Comment by YoKo666** [comment]
Nice to hear it.. :)
Thanx for your support!

**Comment by YoKo666** [comment]
Any news or ETA? :)

**Comment by mx3L** [comment]
Hello, will try to provide some in near future, stay tuned :)

**Comment by YoKo666** [comment]
Looking forward for a new and a fixed version :)

**Comment by mx3L** [comment]
>>Also, if you put option to sync event, if the time is higher than 30 or more minutes for position the titles, it crashes E2.

Can you post crashlog?

**Comment by YoKo666** [comment]
I think this is the right one..i have many and i've deleted this bugged version..[]http://notepad.cc/hinaovu7(url)

**Comment by mx3L** [comment]
Thanks

**Comment by YoKo666** [comment]
No news?

**Comment by mx3L** [comment]
Hello,
As you know this request is here many months now, with not so great results, due to my limited time invested in it and advanced challenge to get timing right with tiresome testing, when rewards are not so great.

I just want you to know that I know about this issue (it's still bothering me :)). I will not make promises like this again
> Hello, will try to provide some in near future, stay tuned :)

since it looks like I'm not doing good estimate.

**Comment by YoKo666** [comment]
:)
Sorry if i am too annoying.. But is there a chance of new version? :)
